test/versioning-strategies/prerelease.ts

```diff
@@ -0,0 +1,280 @@
+import {Version} from '../../src/version';
+
+describe('PrereleaseVersioningStrategyVersioningStrategy', () => {
+  describe('with breaking change', () => {
+    const commits = [
+      {
+        sha: 'sha1',
+        message: 'feat: some feature',
+        files: ['path1/file1.txt'],
+        type: 'feat',
+        scope: null,
+        bareMessage: 'some feature',
+        notes: [],
+        references: [],
+        breaking: false,
+      },
+      {
+        sha: 'sha2',
+        message: 'fix!: some bugfix',
+        files: ['path1/file1.rb'],
+        type: 'fix',
+        scope: null,
+        bareMessage: 'some bugfix',
+        notes: [{title: 'BREAKING CHANGE', text: 'some bugfix'}],
+        references: [],
+        breaking: true,
+      },
+      {
+        sha: 'sha3',
+        message: 'docs: some documentation',
+        files: ['path1/file1.java'],
+        type: 'docs',
+        scope: null,
+        bareMessage: 'some documentation',
+        notes: [],
+        references: [],
+        breaking: false,
+      },
+    ];
+    it('can bump a major', async () => {
+      const strategy = new PrereleaseVersioningStrategy();
+      const oldVersion = Version.parse('1.2.3');
+      const newVersion = await strategy.bump(oldVersion, commits);
+      expect(newVersion.toString()).to.equal('2.0.0');
+    });
+
+    it('can bump a major on pre major for breaking change', async () => {
+      const strategy = new PrereleaseVersioningStrategy();
+      const oldVersion = Version.parse('0.1.2');
+      const newVersion = await strategy.bump(oldVersion, commits);
+      expect(newVersion.toString()).to.equal('1.0.0');
+    });
+
+    it('can bump a minor pre major for breaking change', async () => {
+      const strategy = new PrereleaseVersioningStrategy({
+        bumpMinorPreMajor: true,
+      });
+      const oldVersion = Version.parse('0.1.2');
+      const newVersion = await strategy.bump(oldVersion, commits);
+      expect(newVersion.toString()).to.equal('0.2.0');
+    });
+  });
+
+  describe('with a feature', () => {
+    const commits = [
+      {
+        sha: 'sha1',
+        message: 'feat: some feature',
+        files: ['path1/file1.txt'],
+        type: 'feat',
+        scope: null,
+        bareMessage: 'some feature',
+        notes: [],
+        references: [],
+        breaking: false,
+      },
+      {
+        sha: 'sha2',
+        message: 'fix: some bugfix',
+        files: ['path1/file1.rb'],
+        type: 'fix',
+        scope: null,
+        bareMessage: 'some bugfix',
+        notes: [],
+        references: [],
+        breaking: false,
+      },
+      {
+        sha: 'sha3',
+        message: 'docs: some documentation',
+        files: ['path1/file1.java'],
+        type: 'docs',
+        scope: null,
+        bareMessage: 'some documentation',
+        notes: [],
+        references: [],
+        breaking: false,
+      },
+    ];
+    it('can bump a minor', async () => {
```

Comment:
Again, I'd like to see the impact of features on pre-releases:

- 1.0.0-beta01 => 1.0.0-beta02
- 2.0.0-beta01 => 2.0.0-beta02
- 1.0.1-beta01 => 1.1.0-beta01
- 1.1.0-beta01 => 1.1.0-beta02
- 1.1.1-beta01 => 1.2.0-beta01